kgforge/specializations/models/rdf/rdf_model_service_from_directory.py

```diff
@@ -73,18 +72,31 @@ def _build_shapes_map(self) -> Dict:
                         ?shape a rdfs:Class
                     }
                 }
-            } ORDER BY ?type"""
+            } ORDER BY ?type
+        """
         res = self._graph.query(query)
-        return {row["type"]: row["shape"] for row in res}
 
+        class_being_shaped_id_to_shape_uri: Dict[str, URIRef] = {
+            row["type"]: URIRef(row["shape"])
+            for row in res
+        }
 
-def load_rdf_files(path: Path) -> Graph:
-    memory_graph = Graph()
+        # FIXME should return the file path where the schema is in
+        schema_to_file = dict(
+            (e, "")  # TODO file source
+            for e in class_being_shaped_id_to_shape_uri.values()
+        )
+
+        return schema_to_file, class_being_shaped_id_to_shape_uri
+
+
+def load_rdf_files_into_graph(path: Path, memory_graph: Graph) -> Graph:
```

Comment:
I would not recommend to pass an empty graph and mutate it inside this method. What are you gaining by doing so compare to initialising, building the graph and returning it from this method ?

Reply:
Initially I had called the load_rdf_files_into_graph multiple times, for multiple paths (ontology vs schema). This is no longer the case, but it was in order to reuse the same graph object and load into it files from different locations

Reply:
does it mean that all the files inside a directory will be used to populate the graph?

Reply:
Yes.
In my most recent commit, I get rid of this method completely, so that I can implement the method that was necessary for this [test](https://github.com/BlueBrain/nexus-forge/blob/ab22a830b7d4ed0fc9e01392fe6ff88504a3594b/tests/specializations/models/test_rdf_model.py#L99) to be valid.
It also follows the fix me [here](https://github.com/BlueBrain/nexus-forge/blob/ab22a830b7d4ed0fc9e01392fe6ff88504a3594b/kgforge/specializations/models/rdf/directory_service.py#L35)


I think it's worth looking at the [commit](https://github.com/BlueBrain/nexus-forge/pull/351/commits/8a5dfee3b69abb30c2f16c0c71364c92104cfba3) separately since it's targeting only this feature. I've put an explanation of why I made these changes as a comment in this test.